server/backend/check.ml

```diff
@@ -341,33 +341,37 @@ fi |} pkg pkg pkg (Server_configfile.platform_distribution conf)
 let run_job ~cap ~conf ~pool ~debug ~stderr ~base_obuilder ~extra_repos ~switch ~num logdir pkg =
   Lwt_pool.use pool begin fun () ->
     let* () = Lwt_io.write_line stderr ("["^num^"] Checking "^pkg^" on "^Intf.Switch.switch switch^"…") in
-    let switch = Intf.Switch.name switch in
-    let logfile = Server_workdirs.tmplogfile ~pkg ~switch logdir in
+    let switch_name = Intf.Switch.name switch in
+    let logfile = Server_workdirs.tmplogfile ~pkg ~switch:switch_name logdir in
     let* v =
       Lwt_io.with_file ~flags:Unix.[O_WRONLY; O_CREAT; O_TRUNC] ~perm:0o640 ~mode:Lwt_io.Output (Fpath.to_string logfile) (fun stdout ->
-        ocluster_build ~cap ~conf ~debug ~base_obuilder ~stdout ~stderr (run_script ~conf ~extra_repos pkg))
+        let with_dune = match Intf.Switch.build_with switch with
+        | Intf.Build_with.Dune -> true
+        | Intf.Build_with.Opam -> false
+        in
```

Comment:
```suggestion
        let with_dune =  Intf.Switch.with_dune switch  in
```

Reply:
Haha, good catch. I implemented this like 3 times until I wrote `Intf.Switch.with_dune`.